### ISSUE TITLE
arrayMap(x -> x + 1, [1, 2, 3]) returns wrong result

### ISSUE THREAD

**Issue by l1t1** [quackscience/duckdb-extension-clickhouse-sql#16]
it should return [2,3,4], rather than [1,4,9], others also wrong

```sql
D SELECT arrayMap(x -> x + 1, [1, 2, 3]); 
┌────────────────────────────────────────────────────┐
│ arraymap((x -> (x + 1)), main.list_value(1, 2, 3)) │
│                      int32[]                       │
├────────────────────────────────────────────────────┤
│ [1, 4, 9]                                          │
└────────────────────────────────────────────────────┘
D SELECT arrayMap(x -> x -1, [1, 2, 3]); --also wrong
┌────────────────────────────────────────────────────┐
│ arraymap((x -> (x - 1)), main.list_value(1, 2, 3)) │
│                      int32[]                       │
├────────────────────────────────────────────────────┤
│ [1, 4, 9]                                          │
└────────────────────────────────────────────────────┘
D SELECT arrayMap(x -> x, [1, 2, 3]); --also wrong
┌──────────────────────────────────────────────┐
│ arraymap((x -> x), main.list_value(1, 2, 3)) │
│                   int32[]                    │
├──────────────────────────────────────────────┤
│ [1, 4, 9]                                    │
└──────────────────────────────────────────────┘
```

**Comment by lmangani** [comment]
Thanks for the report! The macro indeed requires a patch to correctly accept and execute the lambda function

**Comment by lmangani** [comment]
Big has been fixed in master and will be part of next release tonight. Thanks for the report @l1t1 

```
D SELECT arrayMap(x -> x + 1, [1, 2, 3]);
┌────────────────────────────────────────────────────┐
│ arraymap((x -> (x + 1)), main.list_value(1, 2, 3)) │
│                      int32[]                       │
├────────────────────────────────────────────────────┤
│ [2, 3, 4]                                          │
└────────────────────────────────────────────────────┘
D SELECT arrayMap(x -> x + 10, [1, 2, 3]);
┌─────────────────────────────────────────────────────┐
│ arraymap((x -> (x + 10)), main.list_value(1, 2, 3)) │
│                       int32[]                       │
├─────────────────────────────────────────────────────┤
│ [11, 12, 13]                                        │
└─────────────────────────────────────────────────────┘
```

Note: the lambda expression syntax requires using `x`

**Comment by lmangani** [comment]
Fixed in chsql v1.0.6 for DuckDB v1.1.3
